service/common/src/main/java/org/apache/polaris/service/context/DefaultRealmIdResolver.java

```diff
@@ -36,15 +37,15 @@ public DefaultRealmIdResolver(RealmContextConfiguration configuration) {
   }
 
   @Override
-  public RealmId resolveRealmContext(
+  public RealmId resolveRealmId(
       String requestURL, String method, String path, Map<String, String> headers) {
 
     String realm;
 
     if (headers.containsKey(configuration.headerName())) {
       realm = headers.get(configuration.headerName());
       if (!configuration.realms().contains(realm)) {
-        throw new IllegalArgumentException("Unknown realm: " + realm);
+        throw new NotAuthorizedException("Unknown realm: " + realm);
       }
     } else {
       realm = configuration.defaultRealm();
```

Comment:
I think it could be valuable to cover this with an explicit config flag. If realms are used in a particular deployment, it is probably not wise to allow potentially malicious clients to try sending requests at the default realm without knowing its ID.

Reply:
I completely agree, I think that having a default realm is a potential security issue.